### PR TITLE
Replace BytesN<32> for contract IDs with Address

### PR DESCRIPTION
**Pull request by leighmcculloch**
### What
Replace BytesN<32> for contract IDs with Address.

### Why
Exposing contract IDs as BytesN<32> is too low level and results in contracts binding themselves to an implementation detail. Contracts are deployed to an Address, and accounts are also an Address, and as much as possile contract developers shouldn't think about these differently, especially once folks start using contracts for authentication.

In all cases we currently use BytesN<32> for a contract ID we can trivially replace it with an Address.

The functions remain for building Addresses from BytesN purely for test capabilities, because in some edge case situations it will be necessary to define a contract ID in a test.

Close https://github.com/stellar/rs-soroban-sdk/issues/868

### CODE REVIEW

File: soroban-sdk/src/env.rs [hunk 1]
```diff
@@ -408,7 +410,7 @@ impl Env {
     {
         let rv = internal::Env::try_call(
             self,
-            contract_id.to_object(),
+            contract_id.contract_id().to_object(),
```
Comment (leighmcculloch) [inline soroban-sdk/src/env.rs:413]: TODO: Catch the error when Address is not a contract. This line is a problem. Try_invoke should never panic, but the call to contract_id() does panic now. I probably should have kept contract_id() as an Option<>. I'll change it back and massage the None into a general error of sorts.

Comment (leighmcculloch) [reply soroban-sdk/src/env.rs]: Fixed in [c02d267](https://github.com/stellar/rs-soroban-sdk/pull/947/commits/c02d2671096a9b475cb6ec0b72faa0389622a961)

**Review by graydon — APPROVED** [review]
I have no real opinion on the ergonomics here, it looks fine to me!
